File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        if len(s) < 2: return len(s)
        verificar = ""
        longo = 0
        for letra in s:
            if letra not in verificar:
                verificar += letra
                longo = max(longo, len(verificar))
            else:
                if verificar[-1] == letra:
                    verificar = letra
                else:
                    verificar = verificar[verificar.index(letra)+1:] + letra
        return longo

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        if (s.length() < 2) return s.length();
        String verificar = "";
        int longest = 0;
        for (int i = 0; i < s.length(); i++) {
            char current = s.charAt(i);

            if (!verificar.contains(String.valueOf(current))) {
                verificar += current;
                longest = Math.max(longest, verificar.length());
            } else {
                if (String.valueOf(current).equals(String.valueOf(verificar.charAt(verificar.length() - 1)))) {
                    verificar = String.valueOf(current);
                } else {
                    int index = verificar.indexOf(String.valueOf(current));
                    verificar = verificar.substring(index + 1) + current;
                }
            }
        }
        return longest;
    }
}